components/latex_inline/latex_inline.tsx

```diff
@@ -0,0 +1,75 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React from 'react';
+import {FormattedMessage} from 'react-intl';
+import type {KatexOptions} from 'katex';
+
+type Katex = typeof import('katex');
+
+type Props = {
+    content: string;
+    enableInlineLatex: boolean;
+};
+
+type State = {
+    katex?: Katex;
+}
+
+export default class LatexInline extends React.PureComponent<Props, State> {
+    constructor(props: Props) {
+        super(props);
+
+        this.state = {
+            katex: undefined,
+        };
+    }
+
+    componentDidMount(): void {
+        import('katex').then((katex) => {
+            this.setState({katex});
+        });
+    }
+
+    render(): React.ReactNode {
+        if (!this.props.enableInlineLatex || this.state.katex === undefined) {
+            return (
+                <span
+                    className='post-body--code inline-tex'
+                >
+                    {'$' + this.props.content + '$'}
```

Comment:
should content have any `$` sign escaped? It's been a long time since I used latex so I'm not 100% sure if it could be a problem

Reply:
I am not sure what you mean with this comment. This line of code is executed when the inline latex setting is dissabled or katex is not yet loaded. So this content is not rendered.

Reply:
oh, but I thought people would be able to select and copy it and that's why you put the `$` signs around, I'm not sure if that's an actual latex delimiter or just some sort of convention

Reply:
The `$` is just added in the text because marked only returns the content of the latex code. So if latex rendering is dissabled then i just added them back in and displayed the latex code.